doc/lua_api.txt

```diff
@@ -3911,12 +3911,16 @@ It can be created via `Settings(filename)`.
 #### Methods
 * `get(key)`: returns a value
 * `get_bool(key)`: returns a boolean
+* `get_np_group(key)`: returns a NoiseParams table
 * `set(key, value)`
     * Setting names can't contain whitespace or any of `="{}#`.
     * Setting values can't contain the sequence `\n"""`.
     * Setting names starting with "secure." can't be set on the main settings object (`minetest.settings`).
 * `set_bool(key, value)`
     * See documentation for set() above.
+* `set_np_group(key, value)`
+	* `value` is a NoiseParams table.
+	* Also, see documentation for set() above.
```

Comment:
Wrong indents.

These replace `minetest.get_noiseparams` and `minetest.set_noiseparams`, this means these two functions are deprecated now?

Reply:
I'm not sure they are deprecated, if they are then it seems to be `minetest.get_mapgen_setting_noiseparams(name)` and `minetest.set_mapgen_setting_noiseparams(name, value, [override_meta])` in 'Environment access' which replace them.

Reply:
The functions you mention change the current active world's settings. `set_noiseparams` writes them to the main configuration (`g_settings`) without touching the active world.

Reply:
So `minetest.get_noiseparams` and `minetest.set_noiseparams` should have been in this settings section of lua_api.txt?

Reply:
Depending on how you would like to group them, they belong to the other mapgen functions or to the settings (like `setting_get_pos()`). For me they're currently both in the right location and don't need movement. The noise params only become relevant when someone wants to use the other mapgen functions in mods.

Reply:
We can't call `minetest.get_noiseparams()` nor `minetest.set_noiseparams()` from `mainmenu`.
```
ERROR[Main]: Main menu error: Runtime error from mod '' in callback handleMainMenuButtons(): ...netest\bin\..\builtin\mainmenu\dlg_settings_advanced.lua:512: attempt to call field 'get_noiseparams' (a nil value)
```
I guess those functions are only for mods?

Reply:
I'm not sure, they seem similar to `minetest.get_mapgen_setting_noiseparams(name)` etc. but hmmmm didn't mark them as deprecated so maybe they have different functionality?